packages/vite-plugin-checker/src/checkers/typescript/main.ts

```diff
@@ -102,16 +102,22 @@ const createDiagnostic: CreateDiagnostic<'typescript'> = (pluginConfig) => {
       // https://github.com/microsoft/TypeScript/pull/33082/files
       const createProgram = ts.createEmitAndSemanticDiagnosticsBuilderProgram
 
-      const host = ts.createWatchCompilerHost(
-        configFile,
-        { noEmit: true },
-        ts.sys,
-        createProgram,
-        reportDiagnostic,
-        reportWatchStatusChanged
-      )
-
-      ts.createWatchProgram(host)
+      if (typeof pluginConfig.typescript === 'object' &&  pluginConfig.typescript.buildMode) {
+        const host = ts.createSolutionBuilderWithWatchHost(ts.sys, createProgram, reportDiagnostic, undefined, reportWatchStatusChanged)
+
+        ts.createSolutionBuilderWithWatch(host, [configFile], {}).build()
```

Comment:
I did some research into TypeScript source code and other code I could find on GitHub which used `createSolutionBuilderWithWatchHost`. I think we should pass `{watch: true}` into the options. https://github.com/microsoft/TypeScript/blob/cecd8c50a076cf9791f7074ec5bce9278fe40fce/src/executeCommandLine/executeCommandLine.ts#L746-L757 

```suggestion
        ts.createSolutionBuilderWithWatch(host, [configFile], { watch: true }).build()
```